packages/core/src/state-management/state-manager.ts

```diff
@@ -80,11 +80,14 @@ export class StateManager {
         takeUntil(timer(timeoutMillis)),
         concatMap((tip) => this._handleTip(state$, tip)),
       )
-      .toPromise().then(() => {
-        if (pinned) {
-          this.syncedPinnedStreams.add(state$.id.toString())
-        }
-      });
+      .toPromise()
+    if (pinned) {
+      this.syncedPinnedStreams.add(state$.id.toString())
+    }
+    // Still lone genesis, no time information in tips, so we do check the signature against _now_.
```

Comment:
We have to be very careful here - what if this `sync()` method never gets called?  For instance, right now if you look at `StateManager.load()`, when you load with `SyncOptions.NEVER_SYNC` the `sync()` method doesn't get called, which leaves a security hole since we never do the signature verification for the genesis commit.

Reply:
good catch.

Reply:
I think we could probably move the `verifyLoneGenesis` call up to `StateManager.load()` and that would probably be a better/more clear place for it.